ERCS/erc-7984.md

```diff
@@ -0,0 +1,291 @@
+---
+eip: 7984
+title: Confidential Fungible Token Interface
+description: An interface for confidential fungible tokens
+author: Aryeh Greenberg (@arr00), Ernesto García (@ernestognw), Hadrien Croubois (@Amxx), Ghazi Ben Amor (@GBAZama), Clement Danjou (@immortal-tofu), Joseph Andre Turk (@jatZama), Silas Davis (@silasdavis), Nicolas Pasquier (@npasquie)
+discussions-to: https://ethereum-magicians.org/t/new-erc-confidential-fungible-token-standard/24735
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-07-03
+---
+
+## Abstract
+
+The following standard defines an implementation of a standard API for confidential fungible tokens via pointers. All amounts in this standard are represented by pointers, whose resolution is implementation specific. The interface defines functions to transfer tokens with pointers, as well as approve operators so the tokens can be transferred by a third-party.
+
+## Motivation
+
+A standard interface allows pointer based confidential tokens on Ethereum to be reused by other applications: from privacy-focused wallets to decentralized exchanges, while keeping transaction amounts private from public view.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119) and [RFC 8174](https://www.rfc-editor.org/rfc/rfc8174).
+
+### Nomenclature
+
+- All amounts in this ERC are pointer based amounts represented by `bytes32` pointers unless otherwise specified. The resolution and manipulation of these pointers is implementation specific.
+
+### Token
+
+### Methods
+
+Compliant tokens MUST implement the following methods, unless otherwise specified:
+
+#### `name()`
+
+Returns the name of the token - e.g. `"MyConfidentialToken"`.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+```solidity
+function name() external view returns (string memory)
+```
+
+#### `symbol()`
+
+Returns the symbol of the token - e.g. `"MCT"`.
+
+OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.
+
+```solidity
+function symbol() external view returns (string memory)
+```
+
+#### `decimals()`
+
+Returns the number of decimals the token uses (e.g. `6`) as a plaintext `uint8`.
+
+```solidity
+function decimals() external view returns (uint8)
+```
+
+#### `confidentialTotalSupply()`
+
+Returns the total token supply.
```

Comment:
1. Which actor/component would legitimately be able to view total supply?
2. If we have one, maybe we should add in "Security Considerations" a warning claiming implementation should prevent deducting balances by querying total supply frequently